cdap-watchdog/src/main/java/io/cdap/cdap/logging/plugins/RollingLocationLogAppender.java

```diff
@@ -295,4 +315,46 @@ public void setFileMaxInactiveTimeMs(long fileMaxInactiveTimeMs) {
   public long getFileMaxInactiveTimeMs() {
     return fileMaxInactiveTimeMs;
   }
+
+  //Implement the class that handles delegation of locationOutputStream to outputStream defined in OutputStreamAppender
+  private class DelegatingOutputStream extends OutputStream {
```

Comment:
Use `java.io.FilterOutputStream`. It already does most of what you need

Reply:
The `java.io.FilterOutputStream` class doesn't allow modifying/setting its protected object 'OutputStream out` once its defined which is required in our case. That is why it wouldn't suffice. Let's discuss this separately.

Reply:
Why? it's a non-final protected field. All you need is a simple child like 
```
class SwitchableOutputStream extends FilterOutputStream {
 public void setDestination(OutputStream destination) {
  this.out = destination;
}
}
```
Are there any problems with this?